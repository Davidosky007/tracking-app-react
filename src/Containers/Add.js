import { useEffect } from 'react';
import PropTypes from 'prop-types';
import { fetchUnits, saveMeasurement } from '../logic/api';
import { loadToken } from '../logic/storage';
import AddCard from '../Components/AddCard';

const Add = (props) => {
  const {
    units, getUnits, values, addValue,
  } = props;
  useEffect(() => {
    const token = loadToken();
    fetchUnits(token).then((units) => {
      getUnits(units);
    });
  }, []);

  const handleSubmit = (unit, id) => {
    const value = values[unit];
    const token = loadToken();
    saveMeasurement(id, value, token);
    addValue(unit, '');
  };

  const handleChange = (unit, value) => {
    addValue(unit, value);
  };

  const unitList = units.map((unit) => (
    <AddCard
      key={unit.id}
      id={unit.id}
      title={unit.title}
      value={values[unit.title]}
      changeHandler={handleChange}
      submitHandler={handleSubmit}
    />
  ));

  return (
    <div className="add-wrapper p-l-20 p-r-20">{unitList}</div>
  );
};

Add.propTypes = {
  units: PropTypes.arrayOf(PropTypes.object).isRequired,
  getUnits: PropTypes.func.isRequired,
  values: PropTypes.shape({}).isRequired,
  addValue: PropTypes.func.isRequired,
};

export default Add;
