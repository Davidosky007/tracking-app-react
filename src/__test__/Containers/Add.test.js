import renderer from 'react-test-renderer';
import Add from '../../Containers/Add';

describe('Add', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Add
      units={[{ id: 1, title: 'unit1' }]}
      getUnits={() => {}}
      values={{ unit1: '1', unit2: '2' }}
      addValue={() => {}}
    />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
