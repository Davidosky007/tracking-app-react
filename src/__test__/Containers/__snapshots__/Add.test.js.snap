// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add renders correctly 1`] = `
<div
  className="add-wrapper p-l-20 p-r-20"
>
  <div
    className="add-card flex space-between align-center m-b-20 p-l-10 p-r-10"
  >
    <div
      className="color-gray medium input-title flex"
    >
      unit1
    </div>
    <input
      className="input"
      id="unit1"
      onChange={[Function]}
      placeholder="0"
      type="number"
      value="1"
    />
    <button
      className="submit-button color-dark-gray medium"
      onClick={[Function]}
      type="submit"
    >
      +
    </button>
  </div>
</div>
`;
